docs/01-app/04-api-reference/01-directives/use-cache.mdx

```diff
@@ -69,20 +69,70 @@ export async function getData() {
+
+## Non-serializable arguments
+
+Any non-serializable arguments, props, or closed-over values will turn into opaque references inside the cached function, and can be only passed through and not inspected nor modified. These non-serializable values will be filled in at the request time and won't become a part of the cache key.
+
+For example, a cached function can take in JSX as a `children` prop and return `<div>{children}</div>`, but it won't be able to introspect the actual `children` object. This allows you to nest uncached content inside a cached component.
+
+```tsx filename="app/ui/cached-component.tsx" switcher
+function CachedComponent({ children }: { children: ReactNode }) {
+  'use cache'
+  return <div>{children}</div>
+}
+```
+
+```jsx filename="app/ui/cached-component.js" switcher
+function CachedComponent({ children }) {
+  'use cache'
+  return <div>{children}</div>
+}
+```
+
+## Return values
+
+The return value of the cacheable function must be serializable. This ensures that the cached data can be stored and retrieved correctly.
+
+## At build time
+
+When used at the top of a [layout](/docs/app/api-reference/file-conventions/layout) or [page](/docs/app/api-reference/file-conventions/page), the route segment will be prerendered, allowing it to later be [revalidated](#during-revalidation).
+
+This means `use cache` cannot be used with [request-time APIs](/docs/app/getting-started/caching-and-revalidating#dynamic-apis) like `cookies` or `headers`.
+
+## At runtime
+
+On the **server**, the cache entries of individual components or functions will be cached in-memory.
+
+Then, on the **client**, any content returned from the server cache will be stored in the browser's memory for the duration of the session or until [revalidated](#during-revalidation).
+
+> In future, Next.js will support storing function cache entries in a persistent cache.
+
+## During revalidation
+
+By default, `use cache` has server-side revalidation period of **15 minutes**. While this period may be useful for content that doesn't require frequent updates, you can use the `cacheLife` and `cacheTag` APIs to configure when the individual cache entries should be revalidated.
+
+- [`cacheLife`](/docs/app/api-reference/functions/cacheLife): Configure the cache entry lifetime.
+- [`cacheTag`](/docs/app/api-reference/functions/cacheTag): Create tags for on-demand revalidation.
+
+Both of these APIs integrate across the client and server caching layers, meaning you can configure your caching semantics in one place and have them apply everywhere.
+
+See the [`cacheLife`](/docs/app/api-reference/functions/cacheLife) and [`cacheTag`](/docs/app/api-reference/functions/cacheTag) API docs for more information.
 
 ## Examples
 
-### Caching entire routes with `use cache`
+### Caching an entire route with `use cache`
```

Comment:
This is a good heading title 👍🏻 Its descriptive and lets us (and the robots) know what the content will be about